velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -38,6 +38,46 @@ namespace facebook::velox::connector::hive {
 
 namespace {
 
+RowVectorPtr makeRowVector(
+    std::vector<std::string> childNames,
+    const std::vector<VectorPtr>& children,
+    int32_t numRows,
+    velox::memory::MemoryPool* pool) {
+  std::vector<std::shared_ptr<const Type>> childTypes;
+  childTypes.resize(children.size());
+  for (int i = 0; i < children.size(); i++) {
+    childTypes[i] = children[i]->type();
+  }
+  auto rowType = ROW(std::move(childNames), std::move(childTypes));
+  return std::make_shared<RowVector>(
+      pool, rowType, BufferPtr(nullptr), numRows, std::move(children));
+}
+
+RowVectorPtr filterPartitionColumns(
+    const std::vector<column_index_t>& partitonCols,
+    const RowVectorPtr& input,
+    velox::memory::MemoryPool* pool) {
+  std::vector<std::string> childNames;
+  std::vector<VectorPtr> childVectors;
+  auto dataColumns = input->childrenSize() - partitonCols.size();
+  childNames.reserve(dataColumns);
+  childVectors.reserve(dataColumns);
+
+  auto type = facebook::velox::asRowType(input->type());
+
+  for (uint32_t i = 0; i < input->childrenSize(); i++) {
+    if (std::find(partitonCols.begin(), partitonCols.end(), i) ==
+        partitonCols.end()) {
+      auto child = input->childAt(i);
+      childNames.push_back(type->nameOf(i));
+      childVectors.push_back(child);
+    }
+  }
+
+  return makeRowVector(
+      std::move(childNames), std::move(childVectors), input->size(), pool);
```

Comment:
For the 4th param, input->pool() is better than passed param to this func.

Reply:
For the 2nd param, would it be okay to pass std::move() to a param of type `const std::vector<VectorPtr>&`?